blockchain/blockchain_test.go

```diff
@@ -1644,3 +1659,400 @@ func TestBlockchain_WriteFullBlock(t *testing.T) {
+		}
+
+		transaction.ChainID = chainIDNumber
+
+		txType := tr["type"].(string) //nolint:forcetypeassert
+
+		txTypeNumber, err := common.ParseUint64orHex(&txType)
+		if err != nil {
+			return nil, err
+		}
+
+		transaction.Type = types.TxType(txTypeNumber)
+
+		transactions = append(transactions, transaction)
+	}
+
+	return &types.FullBlock{Block: &types.Block{Header: header, Transactions: transactions}}, nil
+}
+
+func CustomReceiptsUnmarshall(jsonData []byte) ([]*types.Receipt, error) {
+	var ( //nolint:prealloc
+		dat      map[string]interface{}
+		err      error
+		receipts []*types.Receipt
+	)
+
+	if err = json.Unmarshal(jsonData, &dat); err != nil {
+		return nil, err
+	}
+
+	for _, receiptInterface := range dat["result"].([]interface{}) { //nolint:forcetypeassert
+		receipt := &types.Receipt{}
+		receiptJSON := receiptInterface.(map[string]interface{}) //nolint:forcetypeassert
+
+		receipt.TxHash = types.StringToHash(receiptJSON["transactionHash"].(string)) //nolint:forcetypeassert
+
+		if receiptJSON["contractAddress"] != nil {
+			addr := types.StringToAddress(receiptJSON["contractAddress"].(string)) //nolint:forcetypeassert
+			receipt.ContractAddress = &addr
+		}
+
+		cumulativeGasUsed := receiptJSON["cumulativeGasUsed"].(string) //nolint:forcetypeassert
+
+		receipt.CumulativeGasUsed, err = common.ParseUint64orHex(&cumulativeGasUsed)
+		if err != nil {
+			return nil, err
+		}
+
+		gasUsed := receiptJSON["gasUsed"].(string) //nolint:forcetypeassert
+
+		receipt.GasUsed, err = common.ParseUint64orHex(&gasUsed)
+		if err != nil {
+			return nil, err
+		}
+
+		receipt.LogsBloom = types.Bloom(types.StringToBytes(receiptJSON["logsBloom"].(string))) //nolint:forcetypeassert
+
+		status := receiptJSON["status"].(string) //nolint:forcetypeassert
+
+		statusNumber, err := common.ParseUint64orHex(&status)
+		if err != nil {
+			return nil, err
+		}
+
+		if statusNumber == 1 {
+			success := types.ReceiptSuccess
+			receipt.Status = &success
+		} else {
+			failed := types.ReceiptFailed
+			receipt.Status = &failed
+		}
+
+		var logs []*types.Log
+
+		for _, logInterface := range receiptJSON["logs"].([]interface{}) { //nolint:forcetypeassert
+			log := &types.Log{}
+			logJSON := logInterface.(map[string]interface{}) //nolint:forcetypeassert
+
+			log.Address = types.StringToAddress(logJSON["address"].(string)) //nolint:forcetypeassert
+
+			if logJSON["topics"] != nil {
+				for _, topic := range logJSON["topics"].([]interface{}) { //nolint:forcetypeassert
+					log.Topics = append(log.Topics, types.StringToHash(topic.(string))) //nolint:forcetypeassert
+				}
+			}
+
+			log.Data = types.StringToBytes(logJSON["data"].(string)) //nolint:forcetypeassert
+
+			logs = append(logs, log)
+		}
+
+		receipts = append(receipts, receipt)
+	}
+
+	return receipts, nil
+}
+
+func DirSize(path string) (uint64, error) {
+	var size uint64
+
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		return size, err
+	}
+
+	for _, entry := range entries {
```

Comment:
This can be done a bit simpler using the `filepath.Walk` function:

```go
func dirSize(t testing.T, dir string) (int64, error) {
       t.Helper()
       
	var totalSize int64

	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
		if err != nil {
			return err
		}
		
		if !info.IsDir() {
			totalSize += info.Size()
		}
		
		return nil
	})

	if err != nil {
		return 0, err
	}

	return totalSize, nil
}
```